dbms/src/Flash/Mpp/MPPTaskManager.cpp

```diff
@@ -31,11 +34,47 @@ extern const char random_task_manager_find_task_failure_failpoint[];
 extern const char pause_before_register_non_root_mpp_task[];
 } // namespace FailPoints
 
+namespace
+{
+// Print log for MPPTask which hasn't been removed for over 5 minutes.
+void checkMPPTasks(const std::unordered_map<String, Stopwatch> & monitored_tasks, const LoggerPtr & log)
+{
+    String log_info;
+    for (const auto & iter : monitored_tasks)
+    {
+        auto alive_time = iter.second.elapsedSeconds();
+        if (alive_time >= 300)
+            log_info = fmt::format("{} <MPPTask is alive for {} secs, {}>", log_info, alive_time, iter.first);
+    }
+
+    if (!log_info.empty())
+        LOG_INFO(log, log_info);
```

Comment:
Consider just collect overtimed task info, and do format and logging work outside the lock protection?

Reply:
> Consider just collect overtimed task info, and do format and logging work outside the lock protection?

okk, I will log info outside of the lock